### ISSUE TITLE
Disable tagging when browsing raw mailboxes

### ISSUE THREAD

**Issue by BjarniRunar** [mailpile/Mailpile#1694]
Tagging messages in a mailbox that haven't been added to the main index has no meaning at the moment. We need to update the UI to make this clear, and we need to check the backend to make sure those API calls don't do anything crazy (such as tag the wrong messages).

**Comment by balping** [comment]
I don't know if it's intentional or not, but `Mailpile.instance.tags` [here](https://github.com/mailpile/Mailpile/blob/970415f5733f9ebe97a9791b980ca21a9a41ec03/shared-data/default-theme/html/jsapi/templates/template-modal-tag-picker-item.html#L3)  is undefined as `Mailpile.instance` contains nothing.  Furthermore `Mailpile.instance`  is referenced in other parts.

Just a quick grep:

```
ChangeLog:* Refactoring Mailpile.instance away, fixing misc. buglets
ChangeLog:* Improve composer: autosave fixes, avoid Mailpile.instance
ChangeLog:* Moved JSON Mailpile.instance.tags into jsapi/app.js
default-theme/html/partials/tooltips.html:    <% _.each(tag.subtag_ids, function(stid, key) { var subtag = _.findWhere(Mailpile.instance.tags, {tid: stid}); %>
default-theme/html/partials/head.html:    Mailpile.instance = {};
default-theme/html/jsapi/templates/template-modal-tag-picker-item.html:    <% if (parent) { var parent = _.findWhere(Mailpile.instance.tags, { tid: parent}); console.log(parent); %>
default-theme/html/jsapi/search/ui.js:            var metadata = _.findWhere(Mailpile.instance.metadata, { mid: mid });
default-theme/html/jsapi/message/message.js:      var name = Mailpile.instance.metadata[mid].from.fn;
default-theme/html/jsapi/message/message.js:  var message = Mailpile.instance.messages[mid];
default-theme/html/jsapi/global/activities.js:    source: tagMatcher(Mailpile.instance.tags),
default-theme/html/jsapi/global/activities.js:    source: peopleMatcher(Mailpile.instance.addresses),
default-theme/html/jsapi/compose/events.js:               Mailpile.instance.state.command_url === '/message/') {
external-plugins/forcegrapher/forcegrapher.html:          <% _.each(msg.tag_tids, function(tid) { var tag = _.findWhere(Mailpile.instance.tags, {tid: tid}) %>
mailpile/httpd.py:        # etc), but is unique per Mailpile instance and should leak nothing
mailpile/httpd.py:        #       on a given Mailpile instance, thuse the long TTL and no
mp-virtualenv/share/mailpile/multipile/mailpile-admin.py:        help='Parent web directory for Mailpile instances')
mp-virtualenv/share/mailpile/contrib/forcegrapher/forcegrapher.html:          <% _.each(msg.tag_tids, function(tid) { var tag = _.findWhere(Mailpile.instance.tags, {tid: tid}) %>
mp-virtualenv/share/mailpile/default-theme/html/partials/head.html:    Mailpile.instance = {};
mp-virtualenv/share/mailpile/default-theme/html/partials/tooltips.html:    <% _.each(tag.subtag_ids, function(stid, key) { var subtag = _.findWhere(Mailpile.instance.tags, {tid: stid}); %>
mp-virtualenv/share/mailpile/default-theme/html/jsapi/templates/template-modal-tag-picker-item.html:    <% console.log(parent); if (0) { var parent = _.findWhere(Mailpile.instance.tags, { tid: parent}); console.log(parent); %>
mp-virtualenv/share/mailpile/default-theme/html/jsapi/compose/events.js:               Mailpile.instance.state.command_url === '/message/') {
mp-virtualenv/share/mailpile/default-theme/html/jsapi/message/message.js:      var name = Mailpile.instance.metadata[mid].from.fn;
mp-virtualenv/share/mailpile/default-theme/html/jsapi/message/message.js:  var message = Mailpile.instance.messages[mid];
mp-virtualenv/share/mailpile/default-theme/html/jsapi/global/activities.js:    source: tagMatcher(Mailpile.instance.tags),
mp-virtualenv/share/mailpile/default-theme/html/jsapi/global/activities.js:    source: peopleMatcher(Mailpile.instance.addresses),
mp-virtualenv/share/mailpile/default-theme/html/jsapi/search/ui.js:            var metadata = _.findWhere(Mailpile.instance.metadata, { mid: mid });
mp-virtualenv/lib/python2.7/site-packages/mailpile/httpd.py:        # etc), but is unique per Mailpile instance and should leak nothing
mp-virtualenv/lib/python2.7/site-packages/mailpile/httpd.py:        #       on a given Mailpile instance, thuse the long TTL and no
shared-data/multipile/mailpile-admin.py:        help='Parent web directory for Mailpile instances')
shared-data/default-theme/html/partials/tooltips.html:    <% _.each(tag.subtag_ids, function(stid, key) { var subtag = _.findWhere(Mailpile.instance.tags, {tid: stid}); %>
shared-data/default-theme/html/partials/head.html:    Mailpile.instance = {};
shared-data/default-theme/html/jsapi/templates/template-modal-tag-picker-item.html:    <% if (parent) { var parent = _.findWhere(Mailpile.instance.tags, { tid: parent}); console.log(parent); %>
shared-data/default-theme/html/jsapi/search/ui.js:            var metadata = _.findWhere(Mailpile.instance.metadata, { mid: mid });
shared-data/default-theme/html/jsapi/message/message.js:      var name = Mailpile.instance.metadata[mid].from.fn;
shared-data/default-theme/html/jsapi/message/message.js:  var message = Mailpile.instance.messages[mid];
shared-data/default-theme/html/jsapi/global/activities.js:    source: tagMatcher(Mailpile.instance.tags),
shared-data/default-theme/html/jsapi/global/activities.js:    source: peopleMatcher(Mailpile.instance.addresses),
shared-data/default-theme/html/jsapi/compose/events.js:               Mailpile.instance.state.command_url === '/message/') {
shared-data/contrib/forcegrapher/forcegrapher.html:          <% _.each(msg.tag_tids, function(tid) { var tag = _.findWhere(Mailpile.instance.tags, {tid: tid}) %>
```

I can see in the chenglog, that it has been removed, but  the references have not been updated.

**Comment by BjarniRunar** [comment]
Hi @balping, thanks for reporting. That's not related to this issue, it relates to #879 - the bulk tag tool is completely broken and needs to be fixed.

**Comment by JackDca** [comment]
The commit 114ef22 referenced above appears to have found its way into master. A quick test shows that messages in a mailbox accessed via the Browse button can't be dragged to a tag, nor can the tag be dragged to the message. Also the pile directory is unchanged after that attempted operation. Is more testing needed, or can this issue be closed, or at least removed from the 1.0 blocker list? :)
